### PR TITLE
521 : 📝

### PR DESCRIPTION
**Pull request by JonasAls89**

### CODE REVIEW

File: training/040_projects_and_infrastructure/030_Novice.rst [hunk 1]
```diff
@@ -5,34 +5,60 @@ Novice
...
+    | ├ systems/
+    | ├ variables/
+    | └ node-metadata.conf.json    
+    ├ README.md
+    ├ LICENSE
+    ├ .gitignore
+    └ ++
 
-Dependency tracking hops limit
+The folder ``my-project-directory/`` will be your root folder. You will typically give this folder a name specific to the project and environment you are working on, i.e: ``sesam-development/``. In general, it is recommended to have a total of three Sesam nodes, which in project terms would be ``development/``, ``test/`` and ``production/`` as this ensures CI/CD workflows can be carried out. For this example our environment is development, and as such we will be looking at the ``sesam-development/`` project for now. Finally, in the root folder, you will have files such as README.md, LICENSE and .gitignore. As these files are generic, further description of these will be omitted for now.
```
Comment (ga-hegsvold) [inline training/040_projects_and_infrastructure/030_Novice.rst:41]: The different environments should not have their own folder structure.
Differences in environments should be handled by using environment config files put in the `variables/` folder.

**Review by ga-hegsvold — CHANGES_REQUESTED** [review]

File: training/040_projects_and_infrastructure/030_Novice.rst (hunk 1 above)
Comment (JonasAls89) [reply training/040_projects_and_infrastructure/030_Novice.rst:41]: I'll correct. 👍

File: training/040_projects_and_infrastructure/030_Novice.rst [hunk 2]
```diff
@@ -5,34 +5,69 @@ Novice
...
 
-Mappestruktur
+.. sidebar:: Summary
 
-System
+  The Node config ...
 
-Pipes
+  - is a Sesam node´s skeletal structure
+  - consists of global and local configuration files
```
Comment (ga-hegsvold) [inline training/040_projects_and_infrastructure/030_Novice.rst:16]: Can you elaborate a bit on global vs local config files? Not sure I understand the global-part.
Reading on I think maybe you're referring to `node-metadata.conf.json`? So global meaning global config for a particular Sesam node, right?

File: training/040_projects_and_infrastructure/030_Novice.rst [hunk 3]
```diff
@@ -5,34 +5,69 @@ Novice
...
 
-Pipes
+  - is a Sesam node´s skeletal structure
+  - consists of global and local configuration files   
 
-Node-metadata.conf.json
+The node config in a Sesam node is like its skeletal structure. An example of said skeletal structure is presented below:
 
-+expected
+::
```
Comment (ga-hegsvold) [inline training/040_projects_and_infrastructure/030_Novice.rst:20]: Add a newline here to get proper formatting of the file structure below.

File: training/040_projects_and_infrastructure/030_Novice.rst [hunk 4]
```diff
@@ -5,34 +5,69 @@ Novice
...
+    | | ├ test.json
+    | | └ production.json
+    | └ node-metadata.conf.json    
+    ├ README.md
+    ├ LICENSE
+    ├ .gitignore
+    └ ++
 
-Global metadata
+The folder ``my-project-directory/`` will be your root folder. You will typically give this folder a name specific to the project you are working on, i.e: ``sesam/``. In general, it is recommended to have a total of three Sesam nodes, which in project terms will be development, test and production as this ensures CI/CD workflows can be carried out. Distinct to each environment you will have an environment.json file in your ``variables/`` folder. This will be elaborated on when talking about the ``node/`` folder. Finally, in the root folder, you will have files such as README.md, LICENSE and .gitignore. As these files are generic, further description of these will be omitted for now.
```
Comment (ga-hegsvold) [inline training/040_projects_and_infrastructure/030_Novice.rst:36]: To ensure proper CI/CD workflow we also recommend a dedicated CI node. This is usually a Developer type node (ref _Subscriptions > Products > Choose environment_ inside Sesam).

In general we also recommend that each developer has their own Developer type node, but not sure if that is relevant to mention here.

LICENSE and .gitignore I suppose can be fairly generic, but README.md is usually project specific.
Maybe I'm misunderstanding the intention behind the use of "generic" here?

File: training/040_projects_and_infrastructure/030_Novice.rst [hunk 5]
```diff
@@ -5,34 +5,69 @@ Novice
...
 
-Namespaces
+Moving on from your root folder, the next folder will be your ``node/`` folder. This folder is the one that holds the skeletal information pertaining to all the metadata in your Sesam node. In the ``node/`` folder you will find four subfolders: ``expected/``, ``pipes/``, ``systems/`` and ``variables/`` in addition to the json file ``node-metadata.conf.json``. The subfolders only contain json files. This is an important aspect to remember, as everything in a Sesam node is json.    
 
-Tag for å inkludere c++ ext.
+The ``expected/`` folder has to be created when first downloading the node config via the Sesam-CLI. This approach is described in detail here :ref:`sesam-cli-4-1`. The ``expected/`` folder is used as a folder for all your files used when running tests via the Sesam-CLI. The ``pipes/`` folder contains all your pipe files whilst the ``systems/`` folder contains all your system files. Finally, the ``variables/`` folder will contain all your defined variables files pertaining to each environment. For this example our environment is development, and as such we will be looking at the ``development.json`` file placed in the ``variables/`` folder:
```
Comment (ga-hegsvold) [inline training/040_projects_and_infrastructure/030_Novice.rst:40]: The explanation of the expected/ folder I think can be simplified. Maybe something like:
The `expected/` folder is where all expected outputs from endpoint pipes are stored. These are used by the Sesam-CLI and CI/CD workflows to verify correct outputs.

**Review by ga-hegsvold — CHANGES_REQUESTED** [review]
Got a few comments and questions :-)

File: training/040_projects_and_infrastructure/030_Novice.rst [hunk 6]
```diff
@@ -5,34 +5,71 @@ Novice
...
 
-Mappestruktur
+.. sidebar:: Summary
 
-System
+  The Node config ...
 
-Pipes
+  - is a Sesam node´s skeletal structure
+  - consists of global configuration files and local configuration files
```
Comment (ga-hegsvold) [inline training/040_projects_and_infrastructure/030_Novice.rst:16]: Do we need to distinguish between global and local configs?
Or can we just say that it consists of all configuration files for the Sesam node?

File: training/040_projects_and_infrastructure/030_Novice.rst [hunk 7]
```diff
@@ -5,34 +5,71 @@ Novice
...
 
-Global metadata
+  my-project-directory/
+    ├ node/
+    | ├ expected/
+    | ├ pipes/
+    | ├ systems/
+    | ├ variables/
+    | | ├ development.json
+    | | ├ CI-test.json
```
Comment (ga-hegsvold) [inline training/040_projects_and_infrastructure/030_Novice.rst:29]: I might have been unclear in my comment regarding the CI node.
We don't usually have a separate env.var file for CI. We use the same env.vars as for local dev node.
Ref the `test-env.json` file (default file name the Sesam CLI looks for) that in this example would resides in the `node/` folder (not under `node/variables/`).
Just noticed now that `test-env.json` is missing.

So `CI-test.json` can be removed from `node/variables/ `and `test-env.json` can be added to `node/`.

**Review by ga-hegsvold — CHANGES_REQUESTED** [review]

**Review by ga-hegsvold — APPROVED** [review]

**Pull request merged**